HLTrigger/JetMET/plugins/HLTScoutingCaloProducer.cc

```diff
@@ -104,42 +114,19 @@ HLTScoutingCaloProducer::produce(edm::StreamID sid, edm::Event & iEvent, edm::Ev
 
     //get rho
     Handle<double>rho;
-    if(!iEvent.getByToken(rho_, rho)){
-        edm::LogError ("HLTScoutingCaloProducer") << "invalid collection: rho" << "\n";
-        return;
+    std::auto_ptr<double> outRho(new double(-999));
+    if(iEvent.getByToken(rho_, rho)){
+        outRho.reset(new double(*rho));
     }
-    std::auto_ptr<double> outRho(new double(*rho));
 
     //get MET 
     Handle<reco::CaloMETCollection> metCollection;
-    if(doMet && !iEvent.getByToken(metCollection_, metCollection)){
-        edm::LogError ("HLTScoutingCaloProducer") << "invalid collection: metCollection" << "\n";
-        return;
-    }
-
-    //produce calo jets
-    std::auto_ptr<ScoutingCaloJetCollection> outCaloJets(new ScoutingCaloJetCollection());
-    for(auto &jet : *caloJetCollection){
-        if(jet.pt() > caloJetPtCut && fabs(jet.eta()) < caloJetEtaCut){
-            outCaloJets->emplace_back(
-                    jet.pt(), jet.eta(), jet.phi(), jet.mass(),
-                    jet.jetArea(), jet.maxEInEmTowers(), jet.maxEInHadTowers(),
-                    jet.hadEnergyInHB(), jet.hadEnergyInHE(), jet.hadEnergyInHF(),
-                    jet.emEnergyInEB(), jet.emEnergyInEE(), jet.emEnergyInHF(),
-                    jet.towersArea(), 0.0
-                    );
-        }
-    }
-
-    //produce MET
-    double metPt = -999;
-    double metPhi = -999;
-    if(doMet){
-        metPt = metCollection->front().pt();
-        metPhi = metCollection->front().phi();
+    std::auto_ptr<double> outMetPt(new double(-999));
+    std::auto_ptr<double> outMetPhi(new double(-999));
+    if(doMet && iEvent.getByToken(metCollection_, metCollection)){
+        outMetPt.reset(new double(metCollection->front().pt()));
+        outMetPhi.reset(new double(metCollection->front().phi()));
```

Comment:
@davidlange6 - sorry, I see that this could just be `*outMetPhi = metCollection->front().phi()`.  mistake on my part.